<launch>

  <arg name="x" default="0.0"/>
  <arg name="y" default="0.0"/>
  <arg name="z" default="0.0"/>
  <arg name="yaw" default="0.0"/>
  <arg name="pitch" default="0.0"/>
  <arg name="roll" default="0.0"/>
  <arg name="frame_id" default="velodyne"/>
  <arg name="child_frame_id" default="base_link"/>
  <arg name="period_in_ms" default="100"/>

  <node pkg="tf" type="static_transform_publisher" name="base_link2lidar" args="$(arg x) $(arg y) $(arg z) $(arg yaw) $(arg pitch) $(arg roll) $(arg frame_id) lidar $(arg period_in_ms)"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link2velodyne" args="$(arg x) $(arg y) $(arg z) $(arg yaw) $(arg pitch) $(arg roll) $(arg frame_id) $(arg child_frame_id) $(arg period_in_ms)"/>

  <node type="rviz" name="rviz_elev" pkg="rviz" args="-d $(find elevation_mapping_cupy)/rviz/ramlab_setting.rviz" />

  <!-- Elevation mapping node -->
  <param name="use_sim_time" value="true" />
  <node pkg="elevation_mapping_cupy" type="elevation_mapping_node" name="elevation_mapping" output="screen">
    <rosparam command="load" file="$(find elevation_mapping_cupy)/config/parameters.yaml" />
  </node>


</launch>
